### PR TITLE
HTBHF-1702 Now validating that the confirmation code added is correct…

### PR DESCRIPTION
**Pull request by dwybourn**
Now generating a random six digit code for the confirmation code.
Now validating that the user has entered the correct code.

### CODE REVIEW

File: src/web/routes/application/enter-code/validate.js [hunk 1]
```diff
@@ -1,9 +1,11 @@
+const { CONFIRMATION_CODE_SESSION_PROPERTY } = require('../common/constants')
 const { check } = require('express-validator')
 const { translateValidationMessage } = require('../common/translate-validation-message')
 
+const validateConfirmationCodeIsCorrect = () => (confirmationCode, { req }) => req.body.confirmationCode === req.session[CONFIRMATION_CODE_SESSION_PROPERTY]
+
 const validate = [
-  check('confirmationCode')
-    .not().isEmpty()
+  check('confirmationCode').custom(validateConfirmationCodeIsCorrect())
```
Comment (gethyn1) [inline src/web/routes/application/enter-code/validate.js]: The outer function can be removed here and this line would become:
```
check('confirmationCode').custom(validateConfirmationCodeIsCorrect)
```

File: src/web/routes/application/enter-code/validate.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
+const { CONFIRMATION_CODE_SESSION_PROPERTY } = require('../common/constants')
 const { check } = require('express-validator')
 const { translateValidationMessage } = require('../common/translate-validation-message')
 
+const validateConfirmationCodeIsCorrect = () => (confirmationCode, { req }) => req.body.confirmationCode === req.session[CONFIRMATION_CODE_SESSION_PROPERTY]
```
Comment (gethyn1) [inline src/web/routes/application/enter-code/validate.js]: Could `isCorrect` be removed here, it seems to be implied by `validate`. So function would be called `validateConfirmationCode()`

Comment (dwybourn) [reply src/web/routes/application/enter-code/validate.js]: Updated to `validateConfirmationCode()`

File: src/web/routes/application/enter-code/validate.js (hunk 1 above)
Comment (dwybourn) [reply src/web/routes/application/enter-code/validate.js]: Removed outer function.

**Pull request merged**